sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md

```diff
@@ -1,9 +1,16 @@
 ## Release History
 
-### 4.2.0-beta.1 (Unreleased)
+### 4.2.0
 #### Configuration Changes
 * Changed the default value of `spark.cosmos.read.inferSchema.forceNullableProperties` from `false` to `true` based on user feedback, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22049).
 
+#### Key Bug Fixes
+* Fixes conversion for MapType schema, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22291).
+* Not-nullable properties to include "id", see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22143).
+* Support CustomQuery to be used for inference, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22079).
+* Fixes collision resolution on schema inference, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/21933).
+* Fixes max length of userAgent header, see [PR](https://github.com/Azure/azure-sdk-for-java/pull/22018).
```

Comment:
This should also include a note about determining the micro batch size dynamically now

Reply:
* Improves bulk ingestion throttling rate by dynamically adjusting the max. micro-batch size

Reply:
done thanks for catching this.